lib/appsignal/span.ex

```diff
@@ -135,13 +135,32 @@ defmodule Appsignal.Span do
       |> Appsignal.Span.set_namespace("http_request")
 
   """
-  def set_namespace(%Span{reference: reference} = span, namespace) when is_binary(namespace) do
-    :ok = @nif.set_span_namespace(reference, namespace)
+  def set_namespace(span, namespace) when is_binary(namespace) do
+    set_attribute(span, "appsignal.namespace", namespace)
     span
   end
 
   def set_namespace(span, _name), do: span
```

Comment:
Since `set_attribute` also has pattern matching `%Span{...}` guards set up, and it returns `span` from either implementation, I think we can simplify this to `def set_namespace(span, namespace), do: set_attribute(span, "appsignal.namespace", namespace)

(same for `_if_nil`)